File: api/Woodshed.Application/Models/Response/Identity/UserAccountResponse.cs
namespace Woodshed.Application.Models.Response.Identity;

public class UserAccountResponse
{
    public string Id { get; set; } = string.Empty;
    public string NickName { get; set; } = string.Empty;
    public string Biography { get; set; } = string.Empty;
    public string ImageUrl { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public bool IsFollower { get; set; }
    public bool IsFollowee { get; set; }
    public int FollowersCount { get; set; }
    public int FollowingsCount { get; set; }
}

File: client/src/shared/schemas/response/UserAccountResponse.ts
import { z } from "zod"

export const UserAccountResponseSchema = z.object({
  id: z.string(),
  nickName: z.string(),
  biography: z.string(),
  imageUrl: z.string().nullable(),
  following: z.boolean(),
  followedBy: z.boolean(),
  followersCount: z.number(),
  followingsCount: z.number(),
})

export type UserAccountResponse = z.infer<typeof UserAccountResponseSchema>